### ISSUE TITLE
Audio Crackling in FMVs in Dirt 2 & F1 2009

### ISSUE THREAD

**Issue by Panderner** [hrydgard/ppsspp#12888]
Dirt 2 & F1 2009 Reportedly Audio Crackling in FMV Sequence on these game even the game hits 60 FPS in FMV Sequences.
![Screenshot_2020-05-12-19-57-24-92_2f85358b2198d26f8aca533d68bee793](https://user-images.githubusercontent.com/37503397/81686036-e95ff880-948a-11ea-9721-dc6e0f1a74ed.png)

**Comment by Gamemulatorer** [comment]
related #10865

**Comment by Panderner** [comment]
> related #10865

I've tested these games in the start of this issues works fine I don't notice audio crackling or popping and my phone runs full speed.

**Comment by unknownbrackets** [comment]
A log with "log drop frames" enabled in developer settings would help.

-[Unknown]

**Comment by Panderner** [comment]
> A log with "log drop frames" enabled in developer settings would help.
> 
> -[Unknown]

Logging drop frames is useless for PPSSPP Debugger website. I don't see any dropped frames during FMV Sequence.

**Comment by hrydgard** [comment]
Has this changed in developer builds vs 1.9.3?

**Comment by Panderner** [comment]
> Has this changed in developer builds vs 1.9.3?

same happens in both versions

**Comment by sum2012** [comment]
v1.9.3-830-g9e5be91bf 
Same video\audio issue as #2830 ,but "simulate UMD delay" don't work 
The video is using psmf function
 log: https://drive.google.com/file/d/1VwnfUWvXnTx2xmx3Cv8I80Usf5CqwUnO/view?usp=sharing

**Comment by hrydgard** [comment]
Ok, so this is not a brand new issue at least. Thanks for checking.

Is it like little crackles, or bad choppiness where half the audio is gone? Are the videos running full speed?

**Comment by Panderner** [comment]
> Ok, so this is not a brand new issue at least. Thanks for checking.
> 
> Is it like little crackles, or bad choppiness where half the audio is gone? Are the videos running full speed?

Yes. the FMV Sequence runs full speed but the Audio still Crackling

**Comment by sum2012** [comment]
F1 2009 same issue as Colin McRae : DiRT 2
also using  psmf function
v1.9.3-830-g9e5be91bf log:
https://drive.google.com/file/d/1nvLEwlwRSXylw2bb46TlPcBT8TicvKng/view?usp=sharing

**Comment by sum2012** [comment]
@hrydgard  Do psmfplayer->playMode and psmfplayer->playSpeed are really effect the video speed ?

**Comment by hrydgard** [comment]
I have no idea really. Sounds like a place to start investigating.

**Comment by sum2012** [comment]
Both games are  mode = 0 , speed = 1

**Comment by unknownbrackets** [comment]
IIRC when trying to create tests, I couldn't get playSpeed to make anything different happen.  1 is the typical value.  Mode 0 is the normal mode.

-[Unknown]

**Comment by Panderner** [comment]
here's a YouTube video playing F1 2009 on PPSSPP and should be happened:
https://youtu.be/WdVSu7dY_yQ?t=14s

**Comment by sum2012** [comment]
JPCSPTrace (real psp) on Colin McRae : DiRT 2 of 2 timing

06:27:49.949 user_main - IN  scePsmfPlayerGetVideoData 0x89C13F4, 0x8A45174
06:27:49.949 user_main - OUT scePsmfPlayerGetVideoData 0x89C13F4, 0x8A45174 = 0x8061600C

06:27:50.350 user_main - IN  scePsmfPlayerGetVideoData 0x89C13F4, 0x8A45174
06:27:50.355 user_main - OUT scePsmfPlayerGetVideoData 0x89C13F4, 0x8A45174 = 0x0

06:27:49.943 FmvAudio - IN  scePsmfPlayerGetAudioData 0x89C13F4, 0x960E740
06:27:50.348 FmvAudio - OUT scePsmfPlayerGetAudioData 0x89C13F4, 0x960E740 = 0x0

**Comment by benderscruffy** [comment]
Is this still an issue ???

**Comment by sum2012** [comment]
v1.13.2-2087-g71410f2cb Windows 64 bit
still same issue
 Colin McRae : DiRT  log:
https://gist.githubusercontent.com/sum2012/8cd0c3b1f42991a6d526964da7d6dadd/raw/7ed19f57fcbcd8033f0b9d86d704fd3d7efe69ad/gistfile1.txt

**Comment by hrydgard** [comment]
Try this way: #20200

**Comment by Panderner** [comment]
> Try this way: [#20200](https://github.com/hrydgard/ppsspp/issues/20200)

Nope, #20200 does not fix this.

**Comment by Panderner** [comment]
Slow and crackling FMVs still an issue on latest builds.

**Comment by hrydgard** [comment]
I traced this on hardware using jpcsptrace, and it's clear that there's two threads fighting over sceAudioOutput2OutputBlocking , the FMV thread (with higher priority, but not sure it matters) is managing to enqueue its buffers while the sceSas (audio effect) thread gets SCE_ERROR_AUDIO_CHANNEL_BUSY 

Turns out that this function is not as completely blocking as it sounds, it does limit the enqueues - while in PPSSPP, we simply block. This leads to the movie thread pushing a frame of audio, the effects thread pushing a frame of silence, with the result of the video playing slow and stuttery.

Need to fix the blocking behavior of sceAudioOutput2OutputBlocking.
